### ISSUE TITLE
git-lfs should honor the --quiet flag to the clone command

### ISSUE THREAD

**Issue by Gustra** [github/git-lfs#1270]
Running git-lfs 1.1.2.
This is a minor inconvenience, but when executing "git clone --quiet", no progress should be printed on the stream. From the git clone man page:

       --quiet, -q
           Operate quietly. Progress is not reported to the standard error stream. This flag is also passed to the 'rsync' command when given.

Git lfs should honor this flag too during clone, _if possible_. Currently git-lfs prints progress information regardless of the state of the quiet flag.

**Comment by technoweenie** [comment]
Seems legit :+1:  I wonder if that quiet flag gets propagated down to the smudge filter? Maybe it sets an env var or something?

**Comment by technoweenie** [comment]
To be clear, are you talking about running `git lfs clone`, and the `--quiet` flag being ignored? If so, this should be fixed in v1.2.1, which I hope to ship this week.

If you just mean a regular `git clone`, then refer to my original reply.

**Comment by Gustra** [comment]
Hi,

I am talking about "git clone" not "git lfs clone", to be clear. Also, I
have no idea whether git propagates this setting down the line. If it does
not then I guess a ticket on git would be appropriate too...

BR
Gunnar


2016-06-01 23:23 GMT+02:00 risk danger olson <notifications@github.com>:

**Comment by ttaylorr** [comment]
@Gustra I agree; sounds like we may want to open an issue on Git itself, then.

**Comment by kenyon** [comment]
Same with "git push --quiet". Right now you get the "Uploading LFS objects" progress output, but that should be suppressed if doing "git push --quiet".

**Comment by PhilLab** [comment]
The --quiet option is very convenient when scripting, e.g. in a CI environment. Currently, we get a lot of messages like
```
Filtering content:  12% (108/862), 21.60 MiB | 40.65 MiB/s
```
polluting the logs

**Comment by bk2204** [comment]
The message you're seeing, "Filtering content", actually comes from Git, not Git LFS. I agree that Git should honor `--quiet` in this case, but that issue is better reported to Git upstream. That issue is actually fairly easily fixable by upstream.

More generally, though, Git doesn't pass down the `--quiet` flag to us. So unless Git learns how to tell us that the user has requested quiet operation, we have no way of knowing that and acting on it.

**Comment by vermiculus** [comment]
Sorry to revive an old thread, but if this was reported upstream, can that be linked there? For better or for worse, this is the top result of `git quiet "filtering content"` and I'm having trouble locating the upstream ticket if it exists (at least, on `git-scm.org`).

If other folks can't find it either, I can go ahead and report it.

**Comment by bk2204** [comment]
Git doesn't have a bug tracker; you'd have to send an email (which you could create with `git bugreport` in newer versions) to the mailing list to report a bug.  I have not reported it there and I don't recall seeing it reported there, but you certainly could.

**Comment by vermiculus** [comment]
Submitted https://public-inbox.org/git/CABceR4ZFVW=zeSwef7_dP+TWZ29J7BUkmMEB1CzCz=et_yYS9w@mail.gmail.com/.

**Comment by KalleOlaviNiemitalo** [comment]
That patch suppresses the "Filtering content" message, but it does not add anything that would pass the --quiet option to hooks and filters, to let Git LFS suppress its ["Uploading LFS objects"](https://github.com/git-lfs/git-lfs/blob/7266ee06beb3fe0a641640322dc12bf40ea9dd29/tq/meter.go#L235) message. Is that still missing or has it been done separately?
